UnitySDK/Assets/ML-Agents/Scripts/DemonstrationStore.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.IO;
+using UnityEngine;
+
+
+namespace MLAgents
+{
+    /// <summary>
+    /// Demonstration meta-data.
+    /// </summary>
+    [System.Serializable]
+    public struct DemonstrationMetaData
+    {
+        public int numberExperiences;
+        public int numberEpisodes;
+        public int API;
+    }
+    
+    public class DemonstrationStore
```

Comment:
The current API exposed by DemonstrationStore exposes several implementation details (e.g. creating a directory, managing a file and saving metadata). It should really only contain 3 methods to abstract away that under the hood it is relying on file storage.
- initialize (this handles creating the directory, initializing the file and writing the brain parameters, which can remain as separate, private methods)
- record (is just a rename of WriteExperience)
- close (handles saving metadata)

Reply:
Makes a lot of sense. Will do.